### PR TITLE
[WIP] Groups

### PR DESCRIPTION
**Pull request by dzucconi**
- [x] equal size overlay buttons
- [x] implement user/group blocks
- [x] tighten up group dropdown
- [x] collapsable group section
- [x] Is logged in check for connect/follow buttons
- [x] working follow buttons on channels
- [x] ensure cannot follow own channels
- [x] correct page title ("Are.na / Profile name")
- [x] error pages (user/group not found, group unauthorized)
- [x] fix 'all' view sort
- [x] get random sort working
- [x] Add comments icon to block
- [x] Fix connections on your own profile (support slugs)
- [x] Channel index page
- [x] followed/follower pages
- [x] fix active states for navlinks
- [x] empty states
- [x] Group Avatar uploading / display
- [x] profile tips
- [x] block lightbox triggering
- [x] channels index empty state
- [x] infinite scroll sometimes resets
- [x] User blocks should hover to get follow state
- [x] Some weird issue with my own following page and infinite scroll
- [x] Fix block button clicks from propagating and opening lightbox
- [x] Fix create private channel connect overlay issue
- [x] ensure view/sorts get persisted and used
- [x] Multiple Connect UI's on screen at once causes crash due to searching
- [x] Disable groups in user pulldown
- [x] Disable links to groups
- [x] Disable loading of groups profiles

### QA
- [x] Tips should show on new profiles regardless of whether they have content or not (and the account is under a week old)
- [x] Check filter bar when searching for blocks
- [x] Confirm disable group profiles
- [x] Mobile-sized window: filter bar alignment is off
- [x] Meta tags missing for profile (see old meta.jade file)
- [x] Hide loading circle on cases where the image is a transparent PNG

### Hold (separate PR, move this to a new issue once closed)
- [ ] refresh group on settings save
- [ ] user pulldown needs to stop at bottom margin and scroll when it overflows
- [ ] persist close on group tip in new group modal
- [ ] Persist collapsed groups state in user dropdown
- [ ] Re-enable groups features

### Replicate?
- [ ] often images don't load until hovering (edit: still getting this)

### CODE REVIEW

**Comment by dzucconi** [comment]
@broskoski This branch isn't up to date, just FYI. I'll push a bunch more tomorrow.

**Comment by dzucconi** [comment]
groups-fi

**Comment by csherron** [comment]
** groups QA
 
new channel 
- [ ] put author dropdown in between name and description
- [ ] avatar in group author dropdown (examples here - https://zpl.io/2jwMKe6)
- [ ] author dropdown - add “+ new group” link at the bottom - clicking this link opens up ‘new group’ modal

edit channel
- [ ] in transfer ownership search bar - “search user or group to transfer channel to”

new group modal(https://zpl.io/bJZQK6K)
- [ ] header should say “New group” instead of Create group
- [ ] opening the modal should activate the first form text field
- [ ] when creating a new group the bottom CTA should say ‘create group’ not done
- [ ] add closable helper dialogue (the yellow rectangle in the zeplin sketch)
- [ ] add privacy dropdown — options are “Public group” and “Secret group” (example dropdown copy here - https://zpl.io/2jwMKe6)
- [ ] remove ‘members’ label - put members list directly below the user search bar
- [ ] no form validation happening (for example, if i forget to enter group name, i can still press done, the modal then disappears when there should be an error)
- [ ] more dire message when deleting a group (to come from leo)
- [ ] opt-in collaborator confirmation: when searching for users to invite, the button on their results say ‘invite’ instead of ‘add’ - and in the members list below - the button either says ‘Cancel invitation’ (if they didn’t accept yet), ’Remove’(if they already accepted) or ‘Owner’ (you) 

edit group modal
- [ ] remove ‘members’ label - put members list directly below the user search bar
- [ ] header needs to say “Edit GroupName” instead of just the group name 
- [ ] add ‘transfer channels’ section (bottom right example here: https://zpl.io/an9dBMY) allowing you to search and add channels you own directly to the group

group profile
- [ ] header button - instead of ‘settings’ it should say ‘edit group’ 
- [ ] admins name should be included in members list
- [ ] need all/blocks view modes (channels can still be the default view mode)
- [ ] follow button in the header doesn’t seem to work

personal profiles
- [ ] missing filter bar
- [ ] if user has public groups - then the user profile needs ‘groups’ link underneath followers/following (see here top left - https://zpl.io/VxlNxYJ)

group channel
- [ ] needs ‘created by UserName’ attribution in Info section
- [ ] remove (edit) link on collaborators names in collaborator section - clicking those names should take you to the group, not the edit modal (example top right here - https://zpl.io/VxlNxYJ)
- [ ] group should still show its own group name under the collaborators section (even if the group authored the channel)

connect
- [ ] ‘close’ button font size should match ‘unfollow / connect’ font size and be vertically centered in the divider button

group badge
- [ ] add group badge with correct type/stroke color to all elements specified here: https://zpl.io/VxlNxYJ 
- [ ] group block caption needs a badge (bottom right here https://zpl.io/VxlNxYJ )

notifications dropdown
- [ ] ‘view all notifications’ button needs left/right margins (not full bleed)

block
- [ ] source / connect border radius - try 5px
- [ ] source / connect buttons should each be 130px wide
- [ ] 20px more padding at the bottom of each row of blocks displayed in grid format
- [ ] re-add comments icon, use same border radius as overlay buttons

user block / group block
- [ ] background for users needs to be changed to white 
- [ ] group profiles can use the grey background, still

channel block
- [ ] big channel names - the title text overflows
- [ ] border radius 1 or 2px maybe? soft corners
- [ ] unfollow / connect buttons
    - [ ] vertically center text 
    - [ ] top border of buttons should match its pure privacy color (not the lighter hoverstate color)
    - [ ] each button should be 130px wide 

text block
- [ ] body text in text block should turn black on hover

**Comment by dzucconi** [comment]
### Punting
- [ ] author dropdown - add “+ new group” link at the bottom - clicking this link opens up ‘new group’ modal

File: react/apollo/index.js [hunk 1]
```diff
@@ -50,6 +56,14 @@ export const initApolloClient = ({ token: X_AUTH_TOKEN, currentRoute, isLoggedIn
...
         isLoggedIn,
       },
+      cookies: {
+        __typename: 'Cookies',
+      },
+    },
+    resolvers: {
+      Cookies: {
+        get: (_obj, args) => cookies[args.name] || null,
+      },
```
Comment (dzucconi) [inline react/apollo/index.js]: I do think we should update this using mutations but I don't feel like dealing with it right now.

File: react/apollo/localState/clientData.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const currentRoute = { ...url.parse(CURRENT_URL || window.location.href) };
+  const isLoggedIn = !!(CURRENT_USER && CURRENT_USER.id);
+
+  // TODO: Better method of this:
+  const pairs = document.cookie.split(';');
+  const cookies = {};
+  for (let i = 0; i < pairs.length; i++) {
+    const pair = pairs[i].split('=');
+    cookies[(`${pair[0]}`).trim()] = unescape(pair[1]);
+  }
```
Comment (dzucconi) [inline react/apollo/localState/clientData.js]: Oh woops; forgot to replace this placeholder code.

File: react/components/ProfileTips/fragments/profileTips.js [hunk 3]
```diff
@@ -6,5 +6,9 @@ export default gql`
     id
     has_seen_classic_channels: flag(name: "has_seen_classic_channels")
     has_seen_bookmarklet_message: flag(name: "has_seen_bookmarklet_message")
+    counts {
+      channels
+      blocks
+    }
```
Comment (dzucconi) [inline react/components/ProfileTips/fragments/profileTips.js]: This needs to be pushed to API:production

File: react/components/Cell/components/Konnectable/components/Embed/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+export default class Embed extends Component {
+  static propTypes = {
+    embed: propType(embedFragment).isRequired,
+  }
+
+  render() {
+    const { embed, ...rest } = this.props;
+
+    return (
+      <Generic src={embed.src} alt={embed.title} {...rest} />
```
Comment (broskoski) [inline react/components/Cell/components/Konnectable/components/Embed/index.js]: Change `alt` to `title` here (`Generic` doesn't accept an `alt` prop)

**Comment by dzucconi** [comment]
WELCOME HOME

**Pull request merged**